widget/class-widget.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+// Block direct requests
+if ( !defined('ABSPATH') )
+   die('-1');
+
+/**
+ * Volunteer Management Widget Class
+ */
+class WI_Volunteer_Management_Widget extends WP_Widget {
+
+   /**
+   * Set up widget information
+   */
+   function __construct() {
+
+      // WP_Widget constructor accepts Base ID (string), Name (string), Widget Options (array), Control Options (array)
+      parent::__construct(
+         'WI_Volunteer_Management_Widget', // Base ID
+         __('Volunteer Management Widget', 'text_domain'), // Name
+         array( 'description' => __( 'A list of volunteer opportunities.', 'text_domain' ), ) // Widget Options
+      );
+
+   }
+
+   /**
+   * Front-end display of widget.
+   *
+   * @see WP_Widget::widget()
+   *
+   * @param array $args     Widget arguments.
+   * @param array $instance Saved values from database.
+   */
+   public function widget( $args, $instance ) {
+
+      // Global variables to  be used in templates/opps-list-widget.php
+      global $display_opp_when;
+      global $display_opp_where;
+      global $display_opp_spots;
+
+      isset( $instance['opp_info_when'] ) ? $display_opp_when = true : $display_opp_when = false;
```

Comment:
If you wanted to you could write this like:

`
<?php
$display_opp_when = isset( $instance['opp_info_when'] ) ? true : false;
?>
`

Not a big deal by any means, but may make it a little cleaner.